src/components/charts/timeline.js

```diff
@@ -4,39 +4,75 @@ import React, { Component } from 'react';
 import C3 from 'c3';
 import moment from 'moment';
 import _ from 'lodash';
+import Rx from 'rxjs';
 import Config from '../../common/config';
 
 import './chart.css';
 
+const SLIDING_WINDOW_SIZE = 300;
+
 class Timeline extends Component {
+
   constructor(props) {
     super(props);
-    this.state = {
-      timeline: {}
-    };
+    this.chart = {};
+    this.chartDataQueue = [];
   }
 
   componentDidMount() {
-    this.setState({ timeline: C3.generate({ ...this.props.chartConfig }) });
+    // Every five minutes, rebuild the chart to avoid a memory leak in the chart library

```

Comment:
Can you add a GitHub Issue that maps to this - we should explain that this is a workaround b/c of a bug in D3.